### DISCUSSION TITLE
Agent should report when a nodes hard drive is read-only

### DISCUSSION THREAD

**Discussion by MatthewABrantley** [netdata/netdata#12612] [Ideas]
I have a server with Netdata installed which had it's root drive begin to fail, and so Ubuntu mounted the drive as read-only to avoid further corruption, which seems to be the default behavior of Ubuntu. 

However, Netdata seems unable to determine anything about that and did not report to me what is occurring or flag the node as being in an error state. 

Here is what it looks like, on the left in the terminal I am SSH'd into the node trying to diagnose the issue, on the right you can see that every individual chart is in an error state, but Netdata is not throwing any errors for the node as a whole. 
![netdataROM-noerror](https://user-images.githubusercontent.com/20295023/161815403-9d3d7c1d-f1d9-4aad-89e3-3ba615efa50a.png)

I also am unable to access my nodes local dashboard whatsoever, hitting :19999 on the nodes internal IP gives me nothing, but works just fine for all my other nodes. 
![NoDashOverVPN](https://user-images.githubusercontent.com/20295023/161815765-87c5fb66-f900-4f4c-b7c1-fe8c2b00fddf.png)

I feel that if a node does enter this state, somehow Netdata should be able to determine that, and probably flag it as a Red Error.

**Comment by thiagoftsm** [comment]
Hello @Raltrwx ,

Thank you for your suggestion. Considering that on Linux this information is available in `/proc/mounts`, and Netdata has condition to parse this file, this solution is possible to be implemented.

ping @cpipilas, and @ilyam8 .

**Comment by MatthewABrantley** [comment]
Yeah, that is one file my server can still access and read, and it does reveal the RO status:

/dev/mapper/ubuntu--vg-ubuntu--lv / ext4 ro,relatime,data=ordered 0 0

**Reply by thiagoftsm** [reply]
This is the exact output I am expecting to parse, thank you very much to share with us.

**Comment by cpipilas** [comment]
Thank you @Raltrwx for sharing your thoughts. I like the idea. Would you mind [creating a feature request](https://github.com/netdata/netdata/issues/new?assignees=&labels=feature+request%2Cneeds+triage&template=FEAT_REQUEST.yml&title=%5BFeat%5D%3A+)?

**Comment by MatthewABrantley** [comment]
No problem,

https://github.com/netdata/netdata/issues/12628
